src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java

```diff
@@ -41,11 +55,41 @@ public FetchAndMergeEntry(BibEntry entry, String field) {
      * @param fields - List of fields to get information from, one at a time in given order
      */
     public FetchAndMergeEntry(BibEntry entry, BasePanel panel, List<String> fields) {
+        this.dialogService = panel.frame().getDialogService();
+        this.panel = panel;
+
+        // TODO: Don't run this method as part of the constructor
```

Comment:
Can you resolve this TODO?

Reply:
Yeah, I was just too lazy to do it ;-). Fixed now.